### PR TITLE
Feat(Data Mapper V2): Starting on functions panel

### PR DESCRIPTION
**Pull request by DanielleCogs**
- **Please check if the PR fulfills these requirements**

* [x] The commit message follows our guidelines
* [x] Tests for the changes have been added (for bug fixes/features)
* [x] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature
- **What is the current behavior?** (You can also link to an open issue here)
no functions in v2; three sources of icons- Fluent v8, Fluent v9 and SVG
- **What is the new behavior (if this is a feature change)?**
converted old Function list into new Fluent v9 tree, added new styling and cleaned up old icons- removed all references to Fluent v8 icons and replaced them with new V9 or SVGs
- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no 
- **Please Include Screenshots or Videos of the intended change**:
image

### CODE REVIEW

File: libs/data-mapper-v2/src/components/functionList/FunctionListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { customTokens } from '../../core';
+import type { FunctionData } from '../../models/Function';
+import { getFunctionBrandingForCategory } from '../../utils/Function.Utils';
+import { FunctionIcon } from '../functionIcon/FunctionIcon';
+//import { DMTooltip } from '../tooltip/tooltip';
```
Comment (takyyon) [inline libs/data-mapper-v2/src/components/functionList/FunctionListItem.tsx:5]: remove?

File: libs/data-mapper-v2/src/components/functionList/FunctionListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { Button, Caption1, TreeItem, TreeItemLayout, makeStyles, shorthands, tokens } from '@fluentui/react-components';
+import { useStyles } from './styles';
+
+const fnIconSize = '12px';
+
+const useCardStyles = makeStyles({
```
Comment (takyyon) [inline libs/data-mapper-v2/src/components/functionList/FunctionListItem.tsx:11]: Can we please move the styles to a styles.ts file? imo, keeps it cleaner with tsx file just defining the components

File: libs/data-mapper-v2/src/components/functionList/styles.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  functionTree: {
+    backgroundColor: '#E8F3FE',
+  },
+  functionTreeItem: {
+    backgroundColor: '#E8F3FE',
+    paddingLeft: '10px',
+  },
+});
+
+export const FunctionsSVG = () => (
```
Comment (takyyon) [inline libs/data-mapper-v2/src/components/functionList/styles.tsx:16]: Ahh I had it in the tsx file initially. Can we move it to it's own images.tsx or something please? styles generally just mean the className or styling we apply

File: libs/data-mapper-v2/src/core/state/DataMapSlice.ts [hunk 4]
```diff
@@ -323,42 +323,42 @@ export const dataMapSlice = createSlice({
...
+        //   newState.functionNodes[fnReactFlowKey] = {
+        //     functionData: fnData,
+        //     functionLocations: [state.curDataMapOperation.currentTargetSchemaNode],
+        //   };
+        // }
+
+        // Create connection entry to instantiate default connection inputs
+        //createConnectionEntryIfNeeded(newState.dataMapConnections, fnData, fnReactFlowKey);
+
+        //updateFunctionNodeLocations(newState, fnReactFlowKey);
```
Comment (takyyon) [inline libs/data-mapper-v2/src/core/state/DataMapSlice.ts:357]: please remove the comments if we are not using the code in this PR

File: libs/data-mapper-v2/src/images/FunctionIcons.tsx [hunk 5]
```diff
@@ -20,9 +20,49 @@ const Count32RegularIcon = (props: IconProps) => {
   );
 };
 
+const CreatePercentageIcon = (props: FluentIconsProps) => {
```
Comment (takyyon) [inline libs/data-mapper-v2/src/images/FunctionIcons.tsx:23]: Actually, might want to pull the Functions Icon here?

Comment (DanielleCogs) [reply libs/data-mapper-v2/src/images/FunctionIcons.tsx:23]: I moved it to a separate file. We can keep the icons that represent functions/categories separate.

File: libs/data-mapper-v2/src/utils/Icon.Utils.tsx [hunk 6]
```diff
@@ -163,11 +154,11 @@ export const iconForNormalizedDataType = (
       });
 
       // Null
-      icons = [CircleOff16Regular, CircleOff16Filled];
+      icon = CircleOffRegular;
       break;
     }
   }
-
+  console.log(icon);
```
Comment (ccastrotrejo) [inline libs/data-mapper-v2/src/utils/Icon.Utils.tsx:161]: NIT: console.log

**Review by ccastrotrejo — APPROVED** [review]

**Pull request merged**